driver/node.go

```diff
@@ -327,10 +333,57 @@ func (d *Driver) NodeGetInfo(ctx context.Context, req *csi.NodeGetInfoRequest) (
 // NodeGetVolumeStats returns the volume capacity statistics available for the
 // the given volume.
 func (d *Driver) NodeGetVolumeStats(ctx context.Context, req *csi.NodeGetVolumeStatsRequest) (*csi.NodeGetVolumeStatsResponse, error) {
-	d.log.WithField("method", "node_get_volume_stats").
-		Info("node get volume stats called")
+	ll := d.log.WithField("method", "node_get_volume_stats")
+	ll.Info("node get volume stats called")
 
-	return nil, status.Error(codes.Unimplemented, "")
+	if req.VolumeId == "" {
```

Comment:
Doesn't look like we actually use the `VolumeId`. Are we validating it just because that's what's in the CSI spec?

Reply:
That's correct: the field is [marked as required per the CSI spec](https://github.com/container-storage-interface/spec/blob/4731db0e0bc53238b93850f43ab05d9355df0fd9/spec.md#nodegetvolumestats), and [the upstream unit tests codify this requirement](https://github.com/digitalocean/csi-digitalocean/blob/17343cf952e13ed59c624b0fceef8bed508212c1/vendor/github.com/kubernetes-csi/csi-test/pkg/sanity/node.go#L399).